### ISSUE TITLE
Work out how to start / resume a file stream from a specific point after restart

### ISSUE THREAD

**Issue by robhawkes** [polygon-city/polygon-city-cli#6]
One of the issues with the current streaming approach is that if the process crashes or it restarted then the stream has to start from scratch if it didn't finish processing all the buildings last time – this causes duplication annoyances and all sorts of other pains.

I'm not yet sure if it's possible but it would be great if there was a way to keep track of the location the stream is in the file so it can resume from that point after a restart and not have to re-process / skip all the buildings it has already processed.

There's certainly a way of pausing and resuming an active stream but I don't know if there's a way to start / seek a new stream from a certain point.

To start the ball rolling, the position for seeking could perhaps be found from either:
- Keeping track of some positional identifier from the streaming library / API
- Searching for the last processed building ID and using that line-number as a starting point

**Comment by robhawkes** [comment]
Looks like there are a couple of ways to achieve this:
- Using [ez-streams and the `skip` method](https://github.com/Sage/ez-streams#infinite-streams)
- Using the [`start` and `stop` byte offset with Node.js streams](http://stackoverflow.com/a/21050850/997339) ([docs](https://nodejs.org/api/fs.html#fs_fs_createreadstream_path_options))

**Comment by robhawkes** [comment]
I had a good stab at implementing this and while I managed to set up all the necessary tracking of how many bytes got processed before a crash, the XML streaming modules I'm using throw errors if you start a stream based on an offset (because it thinks the XML is invalid).

Instead, I've implemented a tracking mechanism so the system knows which buildings have already been successfully streamed. If the stream is ever run again, like after a crash, then it will skip any buildings that have already been processed. It's slower than using a byte offset but it's very simple and it works nicely.

I'm going to close this and stop implementing it for now but we can revisit it again in the future if we feel it's needed.
